apps/condo/domains/marketplace/hooks/useCancelStatusModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useCallback, useMemo, useState } from 'react'
+
+import { useIntl } from '@open-condo/next/intl'
+import { Button, Modal, Typography } from '@open-condo/ui'
+import { colors } from '@open-condo/ui/dist/colors'
+
+
+const CancelStatusModal = ({ open, onCancel, onButtonClick }) => {
+    const intl = useIntl()
+    const CancelInvoiceMessage = intl.formatMessage({ id: 'pages.condo.marketplace.invoice.form.cancelInvoiceMessage' })
+    const CancelInvoiceDescription = intl.formatMessage({ id: 'pages.condo.marketplace.invoice.form.cancelInvoiceDescription' })
+    const CancelInvoiceButtonMessage = intl.formatMessage({ id: 'pages.condo.marketplace.invoice.form.cancelInvoiceButton' })
+
+    return (
+        <Modal
+            title={CancelInvoiceMessage}
+            open={open}
+            onCancel={onCancel}
+            footer={[
+                <Button
+                    onClick={onButtonClick}
+                    key='submit'
+                    type='primary'
+                    color={colors.red[5]}
+                >
+                    {CancelInvoiceButtonMessage}
+                </Button>,
+            ]}
+        >
+            <Typography.Text type='secondary'>
+                {CancelInvoiceDescription}
+            </Typography.Text>
+        </Modal>
+    )
+}
+
+type CancelModalPropsType = {
+    onCancel?: () => void
+    onButtonClick?: () => void
```

Comment:
Why this props are not required? What the case of usage CancelModal without callbacks?

Reply:
I think only `onButtonClick` is required. When you click on it there is always some side effect (update the form or Invoice). `onCancel` is not used anywhere yet, deleted.